### ISSUE TITLE
Support for kotlinx-datetime

### ISSUE THREAD

**Issue by mustafaozhan** [Yelp/swagger-gradle-codegen#151]
Hello 👋  It will be great to have support for kotlinx-datetime rather than ThreeTenABP. It will be also required in case you will be supporting Kotlin Multiplatform later on!

**Comment by cortinico** [comment]
Indeed, I've created the `kmp` label so we can track all the things needed to offer a `-kmp` flavor of the template

**Comment by mustafaozhan** [comment]
@cortinico  Another reason now [ThreeTenABp](https://github.com/JakeWharton/ThreeTenABP) has this in their readme.

```
Attention: Development on this library is winding down. Please consider switching to Android Gradle plugin 4.0, java.time.*, and its core library desugaring feature in the coming months.
```

**Comment by cortinico** [comment]
Yup we should definitely look into this 👍 cc @macisamuele
